tests/Modules/SpectatorTargetInfoModule.Tests/Models/CheckpointsGroupTests.cs

```diff
@@ -0,0 +1,92 @@
+﻿using EvoSC.Common.Interfaces.Models.Enums;
+using EvoSC.Common.Models.Players;
+using EvoSC.Common.Util;
+using EvoSC.Modules.Official.SpectatorTargetInfoModule.Models;
+using Xunit;
+
+namespace EvoSC.Modules.Official.SpectatorTargetInfoModule.Tests.Models;
+
+public class CheckpointsGroupTests
+{
+    [Fact]
+    public Task Gets_Player_Data_From_Group()
+    {
+        var checkpointsGroup = new CheckpointsGroup();
+        var targetPlayer = new OnlinePlayer { State = PlayerState.Playing, AccountId = "*fakeplayer1*" };
+
+        checkpointsGroup.Add(
+            new CheckpointData(new OnlinePlayer { State = PlayerState.Playing, AccountId = "*fakeplayer10*" }, 0));
+        checkpointsGroup.Add(new CheckpointData(targetPlayer, 1));
+        checkpointsGroup.Add(
+            new CheckpointData(new OnlinePlayer { State = PlayerState.Playing, AccountId = "*fakeplayer12*" }, 2));
+
+        Assert.Equal(1, checkpointsGroup.GetPlayerCheckpointData(targetPlayer.GetLogin())?.time);
```

Comment:
move "act" statements out of asserts (remember the arrange, act,  assert (AAA) structure)